backend/fms_core/services/sample.py

```diff
@@ -602,12 +601,29 @@ def pool_submitted_samples(samples_info,
         errors.append(f"Reception date is not valid.")
 
     if not errors:
-        # Calculate the total volume of the pool
-        pool_volume = sum(sample["volume"] for sample in samples_info)
+        equal_volume_ratio = True
```

Comment:
Don't do this here ... calculate the ratio outside and pass the actual ratio in ... Keep the business logic outside the services as much as possible ...

Reply:
Isn't the service layer for business logic? Where else would the business logic take place? In the controller layer? model layer?